### PR TITLE
Add 12-bit (raw R12L) compress/decompress support to cmpto_j2k

### PR DESCRIPTION
**Pull request by tmenari**
Hi @MartinPulec,

Using guidance from @benroeder and your previous emails, I have added R12L 12-bit support for Comprimato J2K using a similar approach to cineform. We have done some of our own testing and this is working well for our use-case.

Unlike cineform, the Comprimato SDK supports a 12-bit pixel format, so I am unsure if we need to add a flag to force R12L decoding right now. If the compressed data has fourcc "J2KR", then decoding will be possible to 8/10/12bit depending on output device capabilities.

How does this look to you?

Thanks,
Thomas

### CODE REVIEW

**Comment by MartinPulec** [comment]
Hi Thomas,

I am glad to be in touch with you and thanks for you contribution. AFAIK now the flag is not needed. Currently the pixelformat chosen is the first given by the device as supported for which there exists a decompressor from incoming codecs (eg. given by uv_to_bmd_codec_map for DeckLink).

The reason why such a flag is used for eg. Cineform (and other) is that the content may or may not be decodable to target color space - it may contain YCbCr and decoding to RGB (any bit depth) may not be possible. As J2KR implies RGB color space, it should be ok. Other point is that whenever the decompress indicates ability to decode to 12-bit, it should be able to decode even a codec, that has internally 8-bit or 10 bit.

So to sum up - the reason for that flag is the absence of fine-grained pixel format selection, which defers this to user. However, I am now working on a better solution (compressed codec is introspected first and the most egligible pixel format will be chosen accordingly).

I've quickly looked to the code and it looks very ok. Just one small note - unfortunately now the decoder priorities are honored only if there are 2 decoders to the same pix format (eg. to be able to choose GPUJPEG over FFMPEG), otherwise it is determined by precedence of codecs given by display. Just FYI, it doesn't matter much now (and will be reworked somehow).

I do not have any remarks so if you are content with it, shall I merge now?

Martin

**Comment by tmenari** [comment]
Thanks for getting back in touch. This is working well for us, so if you think it looks OK please go ahead and merge.
